community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/ParameterExpression.scala

```diff
@@ -25,7 +25,13 @@ import pipes.QueryState
 import org.neo4j.cypher.internal.frontend.v3_0.symbols._
 
 case class ParameterExpression(parameterName: String) extends Expression {
-  def apply(ctx: ExecutionContext)(implicit state: QueryState) = state.getParam(parameterName)
+
+  def apply(ctx: ExecutionContext)(implicit state: QueryState) = state.getParam(parameterName) match {
```

Comment:
Perhaps better to filter the parameters on input to the state, instead on fetching from the state.